compiler/ilgen/IlBuilder.cpp

```diff
@@ -1419,6 +1419,77 @@ IlBuilder::Call(const char *functionName, int32_t numArgs, TR::IlValue ** argVal
    return NULL;
    }
 
+/** \brief
+ *     The service is used to atomically increase memory location [\p address + \p offset] by amount of \p value. 
+ *
+ *  \param address
+ *     The starting point of address we are trying to increment.
+ *
+ *  \param offset
+ *     The offset from address.
+ *
+ *  \param value
+ *     The amount of value we are trying to increment for the location.
+ * 
+ *  \note
+ *	   This service is created specifically for Int32.
+ *
+ *  \return
+ *     The old value at the location [\p address + \p offset] after increment.
+ */
+TR::IlValue *
+IlBuilder::AtomicInt32AddWithOffset(TR::IlValue * address, TR::IlValue * offset, TR::IlValue * value)
+   {
+   TR_ASSERT(address->getSymbol()->getDataType() == TR::Address && (offset == NULL || offset->getSymbol()->getDataType() == TR::Address), "First operand must be TR::Address and second operand must be TR::Address or NULL");
+   TR_ASSERT(value->getSymbol()->getDataType() == TR::Int32, "Thle IlType of value must be TR::Int32");
+   TraceIL("IlBuilder[ %p ]::AtomicInt32AddWithOffset (%d, %d, %d)\n", this, address->getCPIndex(), offset == NULL ? 0 : offset->getCPIndex(), value->getCPIndex());
+   appendBlock();
+
+   TR::DataType returnType = TR::Int32;
+   
+   TR::SymbolReference *methodSymRef = symRefTab()->findOrCreateCodeGenInlinedHelper(TR::SymbolReferenceTable::atomicAdd32BitSymbol); //lock add
+   TR::Node *callNode;
+   //Evaluator will handle if it's (address+offset) or address by counting how many nodes
+   callNode = TR::Node::createWithSymRef(TR::ILOpCode::getDirectCall(returnType), offset == NULL ? 2 : 3, methodSymRef);
+   callNode->setAndIncChild(0, loadValue(address));
+   if (offset == NULL)
+      {
+      callNode->setAndIncChild(1, loadValue(value));
+      }
+   else
+      {
+      callNode->setAndIncChild(1, loadValue(offset));
+      callNode->setAndIncChild(2, loadValue(value));
+      }
+
+   genTreeTop(callNode); 
+   TR::IlValue *returnValue = newValue(callNode->getDataType());
+   genTreeTop(TR::Node::createStore(returnValue, callNode)); 
+   
+   return returnValue; 
+   }
+
+/** \brief
+ *     The service is used to atomically increase memory location \p address by amount of \p value. 
+ *
+ *  \param address
+ *     The starting point of address we are trying to increment.
+ *
+ *  \param value
+ *     The amount of value we are trying to increment for the location.
+ *
+ *  \note
+ *     This service is created specifically for Int32.
+ *
+ *  \return
+ *     The old value at the location [\p address + \p offset].
+ */
+TR::IlValue *
+IlBuilder::AtomicInt32Add(TR::IlValue * address, TR::IlValue * value)
```

Comment:
Same comment about the type name appearing in the name of the service. It should be typeless.